### PR TITLE
CSP-105-inner-ConfirmRegulatedStandard

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

**Review by DevenKShah — CHANGES_REQUESTED** [review]
This PR is showing 62 files changed is this correct?

File: src/SFA.DAS.Roatp.Api/Controllers/ProviderCourseEditController.cs [hunk 1]
```diff
@@ -30,5 +31,17 @@ public async Task<IActionResult> Save([FromRoute]int ukprn, [FromRoute]int larsC
             await _mediator.Send(command);
             return NoContent();
         }
+
+        [Route("/providers/{ukprn}/courses/{larsCode}/update-approved-by-regulator")]
+        [HttpPost]
+        public async Task<IActionResult> Save([FromRoute] int ukprn, [FromRoute] int larsCode, ProviderCourseEditConfirmRegulatedStandardModel model)
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Api/Controllers/ProviderCourseEditController.cs:37]: We have agreed to extend existing endpoint for any edits on ProviderCourse. Remove this new endpoint and use the existing one.

**Review by DevenKShah — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Roatp.Application/ProviderCourse/Commands/UpdateApprovedByRegulator/UpdateApprovedByRegulatorCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        public UpdateApprovedByRegulatorCommandHandler(IProviderCourseReadRepository providerCourseReadRepository, IProviderCourseEditRepository providerCourseEditRepository, ILogger<UpdateApprovedByRegulatorCommandHandler> logger)
+        {
+            _providerCourseReadRepository = providerCourseReadRepository;
+            _providerCourseEditRepository = providerCourseEditRepository;
+            _logger = logger;
+        }
+
+        public async Task<Unit> Handle(UpdateApprovedByRegulatorCommand request, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Update confirm regulated standard for ukprn: {ukprn} LarsCode: {larscode}", request.Ukprn, request.LarsCode);
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Application/ProviderCourse/Commands/UpdateApprovedByRegulator/UpdateApprovedByRegulatorCommandHandler.cs:25]: optional: change 'confirm regulated' to 'approved by regulator'

File: src/SFA.DAS.Roatp.Application/ProviderCourse/Commands/UpdateApprovedByRegulator/UpdateApprovedByRegulatorCommandValidator.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+﻿using FluentValidation;
+
+namespace SFA.DAS.Roatp.Application.ProviderCourse.Commands.UpdateApprovedByRegulator
+{
+    public class UpdateApprovedByRegulatorCommandValidator : AbstractValidator<UpdateApprovedByRegulatorCommand>
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Application/ProviderCourse/Commands/UpdateApprovedByRegulator/UpdateApprovedByRegulatorCommandValidator.cs:5]: add test coverage for this validator please

**Review by MarkFCain — COMMENTED** [review]
minor comments and coverage needed on validator

File: src/SFA.DAS.Roatp.Application.UnitTests/SFA.DAS.Roatp.Application.UnitTests.csproj [hunk 4]
```diff
@@ -25,4 +25,8 @@
     <ProjectReference Include="..\SFA.DAS.Roatp.Domain\SFA.DAS.Roatp.Domain.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Application.UnitTests/SFA.DAS.Roatp.Application.UnitTests.csproj:28]: Remove this

**Review by DevenKShah — APPROVED** [review]
Project file needs reverting.

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**